### ISSUE TITLE
TST test_any_non_keyword_deprecation is in the wrong location

### ISSUE THREAD

**Issue by MarcoGorelli** [pandas-dev/pandas#48846]
This test is currently here

https://github.com/pandas-dev/pandas/blob/a59582da8c9fa410ac05e25ee117a7fef996247f/pandas/tests/groupby/test_any_all.py#L64-L83

However, there's no `groupby` in the tested code. So let's move the test just under this one:

https://github.com/pandas-dev/pandas/blob/9289c46e166533dfafd18843137f101624cf41a5/pandas/tests/frame/test_reductions.py#L1333-L1344

**Comment by phofl** [comment]
I don't think this is worth it. We will begin enforcing deprecations soonish anyway. Wdyt?
